core/common/src/main/java/alluxio/security/authorization/AccessControlList.java

```diff
@@ -600,4 +616,71 @@ public String toString() {
     List<String> entries = toStringEntries();
     return String.join(",", entries);
   }
+
+  /**
+   * This is a custom json serializer for AccessControlList class.
+   */
+  public static class AccessControlListSerializer extends StdSerializer<AccessControlList> {
+    private static final long serialVersionUID = -8523910728069876504L;
+
+    /**
+     * Creates a AccessControlListSerializer.
+     */
+    public AccessControlListSerializer() {
+      super(AccessControlList.class);
+    }
+
+    /**
+     * Serialize an AccessControlList object.
+     * @param accessControlList the ACL object to be serialized
+     * @param jsonGenerator json generator
+     * @param serializerProvider default serializer
+     * @throws IOException
+     */
+    @Override
+    public void serialize(AccessControlList accessControlList, JsonGenerator jsonGenerator,
+        SerializerProvider serializerProvider) throws IOException {
+      jsonGenerator.writeStartObject();
+      jsonGenerator.writeStringField("owner", accessControlList.getOwningUser());
+      jsonGenerator.writeStringField("owningGroup", accessControlList.getOwningGroup());
+      jsonGenerator.writeObjectField("stringEntries", accessControlList.toStringEntries());
```

Comment:
This json serializer still calls `toStringEntries`, so it will still be slow? Where is the performance improvement coming from?

Reply:
The performance improvement comes from the saving we get from Thrift serialization/deserialization. Before, we were converting the string entries back to an object and then calling `toThrift` on it. Now we can call `toThrift` directly without parsing the string entries.